ts/lib/typechecking/worker/index.ts

```diff
@@ -31,17 +33,28 @@ export default class TypecheckWorker {
   private lastSettledStatus = this.status;
   private typecheckTimeout?: NodeJS.Timer;
 
-  private projectRoot!: string;
-  private ts!: typeof import('typescript');
-  private watch!: WatchOfConfigFile<SemanticDiagnosticsBuilderProgram>;
-  private compilerOptions!: CompilerOptions;
+  // SAFETY: these are configured during the `start` call, which in turn is run
+  // after construction but *before* anything else happens.
+  private declare projectRoot: string;
+  private declare ts: typeof import('typescript');
+  private declare watch:
+    | WatchOfConfigFile<SemanticDiagnosticsBuilderProgram>
+    | SolutionBuilder<SemanticDiagnosticsBuilderProgram>;
+  private declare compilerOptions: CompilerOptions;
 
   /**
    * Begin project typechecking, loading TypeScript from the given project root.
    */
   public start(projectRoot: string) {
     this.projectRoot = projectRoot;
     this.ts = this.loadTypeScript();
+
+    // In the case of a composite project
+    if (this.compilerOptions.composite) {
```

Comment:
Note: here we don't actually *have* a `this.compilerOptions`. `declare` and `!` alike mislead here.

However, we currently only use `compilerOptions` as a way of handling the `noEmitOnError` setting, which [seems not to be used in composite projects anyway](https://discord.com/channels/@me/689455032629854295/927735143093919775). For that setting, we can refactor the design here to simply match the `tsc` behavior (and document it). That just means making `compilerOptions` optional.

However, as this line itself indicates, we do actually need to load and resolve the project `tsconfig.json` to even know which mode to build in, so I will follow up by introducing *that* here.

This also leads to an obvious question, though: should we be using `noEmitOnError`, which has caused a great deal of back-and-forth and confusion about exactly how it is integrated into the build? Or should we allow users to opt into `ember-cli-typescript`-specific configuration in some way, e.g. via something like an `ember.typescript` settings object in `package.json`?